convert2rhel/main.py

```diff
@@ -54,6 +54,7 @@ def main():
     utils.require_root()
 
     process_phase = ConversionPhase.INIT
+
     # initialize logging
     logger.initialize_logger("convert2rhel.log")
 
```

Comment:
Question:  What do we want to happen when we cannot create the archive?  We explicitly raise IOError here, for instance:

https://github.com/oamg/convert2rhel/pull/359/files#diff-f82efabf7bb126100afd0472e3d8a5902da6576c6ddf75e3184cce9eb647c52aR128

Options I can see are:
* Choose to ignore it, possibly with a warning to the user.  We have the current log. Archiving it is just a convenience for the user.
* Fail with a message instructing the user on how to ix it.  We consider this historical data important enough that we do not want to continue if we cannot protect it.

Reply:
But one thing that's crossed my mind right now is: 

* What are the odds of the current `archive` directory actually being a `file`? 

The way I see it, this directory will be created by us in a specific directory, at least for now. Since we don't handle user-provided paths, is it really necessary for us to do this checking right now? I mean, I see a scenario where this would be really applicable for user-provided paths, but since it's already "hardcoded" in our codebase, is it really necessary for us to do this checking?

Reply:
To @abadger's questions - I'd just print a warning if we are unable to archive an old log as archiving logs is not a critical part of the conversion process, especially when we generate the so called breadcrumbs file https://github.com/oamg/convert2rhel/pull/324 which records the basic information about all executions of convert2rhel.

@r0x0d, instead of checking whether the path is a dir you can add a try/catch block around the `shutil.copyfile` call (catching IOError and OSError to cover both py2 and py3 respectively) and log a warning upon catching an exception. That way you'd catch more issues than just the not-a-dir one, e.g. permission-related issues.

Reply:
We don't necessarily have to check that this specific error condition exists but I think we should catch certain errors at the higher level.  And when we do that, we may want to give the user a good error message.   Let me explain with some pseudo-code:

```
# main.py
try:
    logger.initialize_logger("convert2rhel.log")
except IOError as e:
    print("Unable to initialize the logger: %s" % e)
    print("Stopping the run")
    sys.exit(1)
except ArchiveCreationError as e:
    print("Unable to backup previous log: %s" % e)
    print("Please take care of the issue and re-run convert2rhel")
    sys.exit(1)
```

If we don't have the try, except, then any error will give the user a traceback along with whatever the raw exception message is.  If we raise IOError (and other things that we're calling can also raise IOError), then we can print a generic message and handle all of those the same.  If we raise a different error than what other code could raise, then we can handle different errors in different ways (custom messages, choice to continue for some exceptions and exiting in others, etc).

So, I think we should have a try: except in main.py to catch error conditions having to do with creating the archive and give the user a nicer error message but whether we want to check specifically for `archive` being a `file` depends on what we want to do when that error occurs.  If we want to treat all errors the same, with the same generic message, then we do not need to do the `isdir()` check.  If we want to differentiate the two, then we do need the isdir() check and we need to either raise a unique exception or make sure the IOError we raise is what we want the user to read.

Does that make sense?

Reply:
Catching exceptions at a higher level sounds good. I would probably vote rather for adding the try/except under the `initialize_logger()` to have: a) all things logger contained in the logger.py, and 2) to keep main.py as clean and concise as possible (ideally no conditions or low level code). 

I don't think we need to differentiate between the archive dir path not being a dir and not being able to copy a file into the dir. Because the copying would fail if the path is not a dir. So I'd just remove the isdir() check.


E.g.
```
def initialize_logger(log_name, log_dir=LOG_DIR):
    ...
    try:
        archive_old_logger_files(log_name)
    except IOError as e:
        print("Unable to initialize the logger: %s" % e)
        print("Stopping the run")
        sys.exit(1)
    except ArchiveCreationError as e:
        print("Unable to backup previous log: %s" % e)
        print("Please take care of the issue and re-run convert2rhel")
        sys.exit(1)

        setup_logger_handler(log_name, log_dir)
```

Reply:
We talked about this in chat and decided that `initialize_logger()` is the right place to do the checks (as Michal said, so that we can keep the `main()` function clean and concise) but we don't want to have `print()`s in logger.py.  That's so we can maintain a separation between code which interacts with the user and code which is about having the computer perform actions (it's the difference between code for a library and code for an application).  The way we decided to reconcile those two ideas is to use @bocekm's  version of `initialize_logger()` but move it into main.py.

@r0x0d One thing to note is when you add the exception handling to initialize_logger, leave out the block for ArchiveCreationError.  Since we aren't checking with isdir(), we won't have an opportunity to raise that.